letslambda.py

```diff
@@ -36,89 +31,75 @@
 # add ch to logger
 LOG.addHandler(handler)
 
-def load_config(bucket):
+def load_from_s3(conf, s3_key):
+    """
+    Try to load a file from the s3 bucket and return it as a string
+    Return None on error
+    """
+    try:
+        s3 = conf['s3_client']
+        content = s3.get_object(Bucket=conf['s3_bucket'], Key=s3_key)["Body"].read()
+    except ClientError as e:
+        LOG.error("Failed to load '{0}' in bucket '{1}'".format(s3_key, conf['s3_bucket']))
+        LOG.error("Error: {0}".format(e))
+        return None
+
+    return content
+
+def load_config(s3, s3_bucket, letslambda_config):
     """
     Try to load the letlambda.yml out of the user bucket
     Will return None if the configuration file does not exist
     """
-    conf = bucket.get_key("letslambda.yml")
-    if conf == None:
+
+    try:
+        conf = s3.get_object(Bucket=s3_bucket, Key=letslambda_config)["Body"].read()
+    except ClientError as e:
+        LOG.error("Failed to fetch letslambda configuration '{0}' in bucket '{1}'".format(letslambda_config, s3_bucket))
+        LOG.error("Error: {0}".format(e))
         return None
 
-    confString = conf.read()
-    conf = yaml.load(confString)
-    return conf
+    return yaml.load(conf)
 
-def loadAccountKey(bucket, conf):
+def load_letsencrypt_account_key(conf):
     """
     Try to load the RSA account key from S3. If it doesn't
     succeed, it will create a new account key and try a registration
     with your provided information
+    The letsenrypt account key is needed to avoid redoing the Proof of
+    Possession challenge (PoP). It is also used to revoke an existing
+    certificate.
     """
     LOG.info("Loading account key from s3")
-    key = bucket.get_key("account.key.rsa")
-    newAccountNeeded = False;
-    if key == None:
-        pem = createAndSaveKey(bucket, "account.key.rsa")
+
+    newAccountNeeded = False
+    account_key = load_from_s3(conf, 'account.key.pem')
+    if account_key == None:
+        account_key = create_and_save_key(conf, "account.key.pem", conf['kms_key'])
```

Comment:
Well, all functions are `underscore_case`. You will see variations in variable names since it reuses your original code. Could be made more uniform later.